modules/carrierroute/route_config.c

```diff
@@ -351,10 +351,8 @@ static int backup_config(void) {
 		LM_ERR("out of private memory\n");
 		return -1;
 	}
-	if(strcpy(backup_file, config_file) == NULL){
-		LM_ERR("can't copy filename\n");
-		goto errout;
-	}
+	backup_file[0] = 0;
```

Comment:
I totally agree `strcpy` should not be checked against `NULL`, as `config_file` can never be `NULL`. Enough space should be in `backup_file` pointer as well, so I can't see the reason of checking the return of `strcat` either. But I can't see the reason when "nulling" the `backup_file` might be useful, can you please elaborate?
Personally, in order to sort out this warning/issue, I would drop `strcpy/strcat` in favor of `memcpy`.

Reply:
One concise way of merging both blocks could be using a simple `sprintf(backup_file, "%s.bak", config_file);` - no need to check the result either.  Regarding "nulling" the `backup_file` beforehand, it seems unnecessary to me as well: the `dst` string is just a buffer.

Reply:
It's superfluous: assuming `config_file` could be `NULL` (which is not possible, as it has a default value, so now the script writer must explicitly set it to `NULL` from script - but that's illegal), then the code will crash anyway much earlier, on the very 1st `strlen(config_file)` on line 350.

So @ihsinme: could you rewrite both of those code blocks into a simple `sprintf()`, since the buffer is sufficiently large to hold all data?  Then we can merge it.  Thanks!

Reply:
@liviuchircu 
look please

Reply:
@liviuchircu look please